### ISSUE TITLE
Use pointer cursor for interactive elements

### ISSUE THREAD

**Issue by alexdln** [npmx-dev/npmx.dev#1760]
## General
We have two main positions regarding the cursor:
* the historical context of the W3C (_use pointer cursor for links only_) and the system cursor on devices;
* the established user experience and the standard supported by most sites (_including Google and Wikipedia_).

Last time we gave the default cursor a chance, and at the moment it is written as the rule in our contributing guide.

## Standards

To begin with, what we consistently agree on is that we need a clear and understandable standard and unification of rules. Also it must comply with a11y standards.

Regarding the a11y standards, we were unable to find clear wording for all standards, and in fact, this is clearly stated only in the W3C ([spec](https://www.w3.org/TR/css-ui-4/#valdef-cursor-pointer)) in its original version*

In terms of usage, there is also no absolute standard - most sites prefer pointer, OS almost always use default, some libraries use default

\* W3C recently updated their standard and specified that pointer can be used for interactive elements ([Video](https://www.youtube.com/watch?v=BdNFBgnzbIY), [working group discussion](https://github.com/w3c/csswg-drafts/issues/1936#issuecomment-420722704)), but apparently as a result of wrong changes this line was removed ([commit](https://github.com/w3c/csswg-drafts/commit/1a57a4025c8778dab45ef8d5fbd482e4d6c4ebce#r177842392))

## Internal talks

Even though we want to follow the standard (and given that the working group has decided that using pointer is acceptable) - we also take the established user experience as a key focus. And it almost unequivocally indicates that we're more accustomed to buttons on a website with a pointer.

This is supported by our personal experience as users and as the target group of our website. In terms of accessibility, we must also take into account existing practices, as changing them could make daily experience worse for some users.

## Result

Based on all this, **we're returning the pointer cursor** for **all** interactive elements (_without any distinction or definition of what a link is and what a button is_). This needs to be updated in the contributing guide and updated throughout our site.

**Comment by graphieros** [comment]
Note on how to enable cursor pointer on chart components.

### Add `useCursorPointer` attr in charts configurations

```ts
const chartConfig = computed<VueUiXyConfig>(() => {
  return {
    //...
    chart: {
      //...
      userOptions: {
        //...
        useCursorPointer: true
      }
    }
  }
})
```
List of components on which to apply the change in the  chart config's `userOptions` object:
- TrendsChart.vue
- TimelineChart.vue
- VersionDistribution.vue
- FacetBarChart.vue
- FacetScatterChart.vue

**Comment by pauliecodes** [comment]
@alexdln Does this mean that we want the buttons in the `connect` button/dropdown, including the dropdown trigger itself, to have a `cursor-pointer`? I noticed today that it doesn't have one, and I was considering opening an issue, but after seeing this and knowing there was a lot of talk about the topic, I thought I would ask first.

https://github.com/user-attachments/assets/b66d3dd6-9385-4d25-9a28-b42c0a89e17d

**Comment by alexdln** [comment]
@pauliecodes yes, but we're still using default right now, so it's still correct. But with the update, there will also be a pointer
